specification.json

```diff
@@ -629,6 +636,97 @@
             "content": "The hook MUST NOT alter the `hook hints` structure.",
             "RFC 2119 keyword": "MUST NOT",
             "children": []
+        },
+        {
+            "id": "Requirement 5.1.1",
+            "machine_id": "requirement_5_1_1",
+            "content": "The `provider` MAY define a mechanism for signaling the occurrence of one of a set of events, including `PROVIDER_READY`, `PROVIDER_ERROR`, `PROVIDER_CONFIGURATION_CHANGED` and `PROVIDER_STALE`, with an `provider event details` payload.",
```

Comment:
Discussion - Doubt about the wording `MAY` & the contract agreed between a Provider and the API. 

For events to be subscribed and unsubscribed, I see the need of Provider to expose a method to add event subscriptions (ex:- `providerA.addHandler(Event.TypeA, HandlerImpl)`). I think spec must enforce this, so that API can work correctly.

Reply:
The idea is that events are still optional. If there's nothing for the SDK internals to subscribe to, there will simply be no asynchronous events (such as `FLAG_CONFIGURATION_CHANGED`) fired.

The only events that are guaranteed to fire in any case, are the `PROVIDER_READY` and `PROVIDER_ERROR` events, if the `initialization` succeeds or fails, respectively.

See here: https://github.com/open-feature/spec/pull/182/files#diff-daf30d840f7e7b1752899ac2c4b268af214e6c8e7058549c8d4218abd2e86ba3R104-R107